lib/propolis/src/enlightenment/hyperv/mod.rs

```diff
@@ -533,7 +575,18 @@ impl MigrateSingle for HyperV {
         Ok(migrate::HyperVEnlightenmentV1 {
             msr_guest_os_id: inner.msr_guest_os_id_value,
             msr_hypercall: inner.msr_hypercall_value.0,
-            msr_reference_tsc: inner.msr_reference_tsc_value.0,
+            reference_tsc: match inner.reference_tsc {
+                ReferenceTsc::Disabled => None,
+                ReferenceTsc::Uninitialized => {
+                    panic!("asked to export an uninitialized TSC enlightenment")
```

Comment:
i can tell this is unreachable: to migrate you have to have a `VmObjects`, but to have gotten a `VmObjects` you have to have gotten through `machine::Builder::finalize` which would have moved this out of `Uninitialized`. i don't quite see what would happen if we were asked to migrate out while still `Creating`, but i assume it's either an error or blocks until the VM is created?

either way, seems like this could be `unreachable!` with some gesturing at the state machine in a comment here too